### PR TITLE
elasticsearch trace exporter

### PR DESCRIPTION
**Pull request by malike**
added elasticsearch trace exporter for more intuitive search of traces and easy visualization in kibana

### CODE REVIEW

File: exporters/trace/elasticsearch/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+compile 'io.opencensus:opencensus-api:0.13.0'
+compile 'io.opencensus:opencensus-exporter-trace-elasticsearch:0.13.0'
+runtime 'io.opencensus:opencensus-impl:0.13.0'
+```
+
+#### Register the exporter
+
+The ElasticsearchConfig is the configurations required by the exporter.
+
+      private final static String ELASTIC_SEARCH_URL= "http://localhost:9200";
```
Comment (songy23) [inline exporters/trace/elasticsearch/README.md]: nit: please put this code snippet in a code block:
```java
...
```

File: exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.elasticsearch;
+
+public class ElasticsearchConfiguration {
```
Comment (songy23) [inline exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchConfiguration.java]: Suggestion: consider using `AutoValue.Builder` to implement this class. Example: [StackdriverTraceConfiguration](https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverTraceConfiguration.java)

File: exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchTraceExporter.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      this.elasticsearchConfiguration = elasticsearchConfiguration;
+      StringBuilder sb = new StringBuilder();
+      sb.append(elasticsearchConfiguration.getElasticsearchUrl()).append("/");
+      sb.append(elasticsearchConfiguration.getElasticsearchIndex()).append("/");
+      sb.append(elasticsearchConfiguration.getElasticsearchType()).append("/");
+      indexUrl = new URL(sb.toString());
+      appName = elasticsearchConfiguration.getAppName();
+    }
+
+    static List<String> convertToJson(Collection<SpanData> spanDataList) {
```
Comment (songy23) [inline exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchTraceExporter.java]: suggestion: consider moving these static methods to another util class.

File: exporters/trace/elasticsearch/README.md (hunk 1 above)
Comment (sebright) [reply exporters/trace/elasticsearch/README.md]: Should this be ` ```java`?

**Comment by songy23** [comment]
Also please run [google java formatter](https://github.com/census-instrumentation/opencensus-java/blob/master/CONTRIBUTING.md#style).

**Comment by adriancole** [comment]
Maybe this is the case, but.. is it impossible for census to write data the
needs json escaping? If so, stringbuilder should be fine. Otherwise
whatever data might should be json escaped.

**Comment by malike** [comment]
@songy23 @sebright @adriancole thanks for the feedback.

I do have the [google java formatter](https://github.com/census-instrumentation/opencensus-java/blob/master/CONTRIBUTING.md#style) imported into IntelliJ. I don't know why it's still not applying the style to some of classes. I'll do it manually then.

**Comment by sebright** [comment]
@malike Have you tried running `./gradlew goJF` to format the whole project?

**Comment by malike** [comment]
@sebright Yes and also made the recommended changes. I'll send a PR today

**Pull request closed without merging**